plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/model/virtual/DBVEntity.java

```diff
@@ -483,18 +483,37 @@ public Collection<DBSEntityAttribute> getDescriptionColumns(DBRProgressMonitor m
         return getDescriptionColumns(monitor, entity, descriptionColumnNames);
     }
 
+    @NotNull
+    public <T extends DBSAttributeBase> Collection<T> getDescriptionColumns(@NotNull Collection<? extends T> attributes) {
+        return getDescriptionColumns(attributes, descriptionColumnNames);
+    }
+
     public static Collection<DBSEntityAttribute> getDescriptionColumns(DBRProgressMonitor monitor, DBSEntity entity, String descColumns)
         throws DBException {
         if (CommonUtils.isEmpty(descColumns)) {
             return Collections.emptyList();
         }
-        List<DBSEntityAttribute> result = new ArrayList<>();
         Collection<? extends DBSEntityAttribute> attributes = entity.getAttributes(monitor);
+        if (CommonUtils.isEmpty(attributes)) {
+            return Collections.emptyList();
+        }
+        return getDescriptionColumns(attributes, descColumns);
+    }
+
+    @NotNull
+    public static <T extends DBSAttributeBase> Collection<T> getDescriptionColumns(
+        @NotNull Collection<? extends T> attributes,
+        @NotNull String descColumns
+    ) {
+        if (CommonUtils.isEmpty(descColumns)) {
+            return Collections.emptyList();
+        }
```

Comment:
Redundant second empty check

Reply:
No, it's not. During the first check, the collection might be `null`. During the second check, we verify it has elements.

Reply:
In both situation we return an empty list, Maybe it better to make it nullable and do only one check instead?